rfcs/0039-minaform-tool-design.md

```diff
@@ -0,0 +1,120 @@
+    - Resolved with helm linting and rendering in buildkite to ensure the charts are usable. Should add `helm test` to that pipeline eventually as well.
+10) Passing flags to the daemon process directly is difficult to debug/handle/manage
+    -  Expose command line flags genericly as a list of command line arguments in helm, allowing deployments to replace the default flags as needed and clearly specify them.
+    - Resolved by having the helm accept a list of arguments to the daemon, provide this functionality through `.tfvars`.
+
+## Detailed design
+[detailed-design]: #detailed-design
+
+One of the major design goals is to introduce a set of default terraform network configurations for qa and production that will be the starting point for all networks we deploy, and will be maintained with PR's and strict code review, while still allowing any further modification to fit the needs of any individual deployment, testing, or debugging effort. The design revolves around a improvements to the existing coda-network tool and a few new bash scripts to address specific needs expressed by nathan, brandon, and andrew.  The entirety of mina-automation can then be bundled in a container alongside its dependencies for easy cross-platform use as soon as possible.
+
+**Unlike in the original `minaform` design, none of these scripts are used to actually deploy or manage cluster resources. Terraform will be the only tool for starting or stopping networks, coda-network and these helper scripts just help get to that stage faster.**
+
+### New coda-network features
+
+- `keyset deploy` command for uploading keys as secrets to kubernetes to deprecate scripts/testnet-keys.py.
+    - If we go this direction permanently, it could be integrated as a local-exec in terraform. See the below keypair download initContainer for another approach.
+- Keyset, keypair, and genesis download to allow easy retrieval of the keys required for a given genesis ledger, or to retrieve any specific key from google cloud storage.
+    - Tracking issue: https://github.com/MinaProtocol/mina/issues/6550
+    - The mina-automation docker image w/ `coda-network keypair download` would allow for an initContainer in kubernetes to grab a key for a specific container, removing the need for `keyset deploy` entirely.
+- Using ocaml generate-keypair binary or docker image to generate keys instead of relying on js library.
+    - Tracking issue: https://github.com/MinaProtocol/mina/issues/6548
+- More complete genesis-create command with support for vesting schedules and a yaml-based config file for defining genesis ledger requirements.
+    - Tracking Issue: https://github.com/MinaProtocol/mina/issues/6551
+    - Doing this properly is worthwhile, but a combination of bash + jq + the existing `coda-network genesis` command have unblocked deployments and will be used until a complete design + implementation is ready.
+
+### ./scripts/create-network.sh
```

Comment:
So I somewhat get the use of a wrapper script around creating adhoc networks (i.e. not *prod/public-release*, *qa* etc we would maintain a `tfvars` file for) though have you thought about Terraform [interactive variable input](https://www.terraform.io/docs/commands/apply.html#input-true) as a means to provide inputs like testnet name, daemon/archive images etc?